BLL/Helpers/UserModification/UserModifier.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using BLL.Helpers.Queries.Interfaces;
+using BLL.Helpers.UserUpdating.Interfaces;
+using BLL.Models;
+using DAL.DBModels;
+using DAL.Repositories.Interfaces;
+using Stripe;
+
+namespace BLL.Helpers.UserUpdating
+{
+    public class UserModifier: IUserModifier
+    {
+        private readonly ITransactionRepository _paymentRepository;
+        private readonly IUserRepository _userRepository;
+        private readonly IUserQueryCreator _userQueryCreator;
+
+        public UserModifier(ITransactionRepository paymentRepository, IUserRepository userRepository, IUserQueryCreator userQueryCreator)
+        {
+            _paymentRepository = paymentRepository;
+            _userRepository = userRepository;
+            _userQueryCreator = userQueryCreator;
+        }
+
+        public async Task<UserDTO> GetOrCreateUser(PaymentModel payment)
+        {
+            return  (await _userRepository.GetUser(await _userQueryCreator.GetUser(payment.Email))).LastOrDefault() ?? await CreateUser(payment);
```

Comment:
> When email == null, extra call to db is made, and 2ndly later the code is trying to create user in an external system using null email value

I verify this model in controller. But I think it will be OK if i throw NotValidModelException when email == null and my Middleware will process it